presto-main/src/test/java/com/facebook/presto/security/TestFileBasedSystemAccessControl.java

```diff
@@ -139,11 +201,108 @@ public void testViewOperations()
+        transaction(transactionManager, accessControlManager)
+                .execute(transactionId -> {
+                    accessControlManager.checkCanSetUser(KERBEROS_REALM, "principal");
+                    accessControlManager.checkCanSetUser(KERBEROS_SIMPLE, "principal");
+                    accessControlManager.checkCanSetUser(null, "principal");
+                    accessControlManager.checkCanSetUser(LDAP_PRINCIPAL, "principal");
+                    accessControlManager.checkCanSetUser(null, "alice");
+                    accessControlManager.checkCanSetUser(LDAP_PRINCIPAL, "alice");
+                });
+
+        accessControlShouldFail(
+                accessControlManager,
+                transactionId -> {
+                    accessControlManager.checkCanSetUser(KERBEROS_HOST_REALM, "principal");
+                });
+
+        accessControlShouldFail(
+                accessControlManager,
+                transactionId -> {
+                    accessControlManager.checkCanSetUser(KERBEROS_IP_REALM, "principal");
+                });
+
+        accessControlShouldFail(
+                accessControlManager,
+                transactionId -> {
+                    accessControlManager.checkCanSetUser(KERBEROS_HOST, "principal");
+                });
+
+        accessControlShouldFail(
+                accessControlManager,
+                transactionId -> {
+                    accessControlManager.checkCanSetUser(KERBEROS_HOST_REALM, "alice");
+                });
+
+        accessControlShouldFail(
+                accessControlManager,
+                transactionId -> {
+                    accessControlManager.checkCanSetUser(KERBEROS_IP_REALM, "alice");
+                });
+
+        accessControlShouldFail(
+                accessControlManager,
+                transactionId -> {
+                    accessControlManager.checkCanSetUser(KERBEROS_REALM, "alice");
+                });
+
+        accessControlShouldFail(
+                accessControlManager,
+                transactionId -> {
+                    accessControlManager.checkCanSetUser(KERBEROS_HOST, "alice");
+                });
+
+        accessControlShouldFail(
+                accessControlManager,
+                transactionId -> {
+                    accessControlManager.checkCanSetUser(KERBEROS_SIMPLE, "alice");
+                });
+    }
+
+    private void checkCanSetWithAllPrincipals(AccessControlManager accessControlManager, String user)
+    {
+        accessControlManager.checkCanSetUser(KERBEROS_HOST_REALM, user);
+        accessControlManager.checkCanSetUser(KERBEROS_IP_REALM, user);
+        accessControlManager.checkCanSetUser(KERBEROS_REALM, user);
+        accessControlManager.checkCanSetUser(KERBEROS_HOST, user);
+        accessControlManager.checkCanSetUser(KERBEROS_SIMPLE, user);
+        accessControlManager.checkCanSetUser(null, user);
+        accessControlManager.checkCanSetUser(LDAP_PRINCIPAL, user);
```

Comment:
The only options are Kerberos, LDAP, and null -- if no authentication is used, the input will be null.